src/site/content/en/blog/wakelock/index.md

```diff
@@ -0,0 +1,264 @@
+
+Check out the [Wake Lock demo][demo] and [source][demo-source] for the demo.
+
+### Wake lock types {: #wake-lock-types }
+
+The Wake Lock API provides two types of wake locks, `screen` and `system`.
+While they are treated independently, one may imply the effects of the other.
+For example, a screen wake lock implies that the app should continue running.
+
+#### `screen` wake lock
+
+A `screen` wake lock prevents the device's screen from turning
+off so that the user can see the information that's displayed on screen.
+
+#### `system` wake lock
+
+{% Aside 'caution' %}
+The `system` wake lock is not currently implemented.
+{% endAside %}
+
+A `system` wake lock prevents the device's CPU from entering standby mode so
+that your app can continue running.
+
+### Get a wake lock {: #get-wake-lock }
+
+To request a wake lock, you need to call the `WakeLock.request()` method
+that lives on the `window` object. You pass it the desired wake lock type as
+the first parameter, which *currently* is limited to just `'screen'`. In
+addition, you also need a way to abort the wake lock, which works through the
+generic `AbortController` interface. Therefore, you first create a new
+[`AbortController`](https://developer.mozilla.org/en-US/docs/Web/API/AbortController),
+and then pass the controller's
+[`AbortSignal`](https://developer.mozilla.org/en-US/docs/Web/API/AbortSignal)
+as the second parameter to `WakeLock.request()`. Two things can happen next
+that you need to `catch`:
+
+* The wake lock can, after a while, just be regularly aborted, which you
+  detect by checking if the exception's name is `'AbortError'`. In this
+  context, `AbortError` is actually not an error in the common sense, but
+  just the way `AbortController` works.
+* The browser can also refuse the request for different reasons, for example,
+  because the battery charge level is too low. In this case, the exception's
+  message will contain more details.
+
+```js
+const wakeLockCheckbox = document.querySelector('#wakeLockCheckbox');
+
+if ('WakeLock' in window) {
+  let wakeLock = null;
+
+  const requestWakeLock = () => {
+    const controller = new AbortController();
+    const signal = controller.signal;
+    window.WakeLock.request('screen', {signal})
+        .catch((e) => {
+          if (e.name === 'AbortError') {
+            wakeLockCheckbox.checked = false;
+            console.log('Wake Lock was aborted');
+          } else {
+            console.error(`${e.name}, ${e.message}`);
+          }
+        });
+    wakeLockCheckbox.checked = true;
+    console.log('Wake Lock is active');
+    return controller;
+  };
+
+  wakeLockCheckbox.addEventListener('change', () => {
+    if (wakeLockCheckbox.checked) {
+      wakeLock = requestWakeLock();
+    } else {
+      wakeLock.abort();
+      wakeLock = null;
+    }
+  });
+}
+```
+
+### The wake lock lifecycle {: #wake-lock-lifecycle }
+
+When you play with the [wake lock demo][demo], you'll notice that wake locks
```

Comment:
This paragraph had a lot of complex sentences that took me a moment to parse. I tried to reduce syntactic complexity in my commit, but feel free to revert if it introduced any problems.

Reply:
I like the clean up, I added one additional clarification.